Sources/Subs-Db-postgresql.php

```diff
@@ -202,6 +202,13 @@ function smf_db_replacement__callback($matches)
 				smf_db_error_backtrace('Wrong value type sent to the database. Date expected. (' . $matches[2] . ')', '', E_USER_ERROR, __FILE__, __LINE__);
 		break;
 
+		case 'time':
+			if (preg_match('~^([0-1]?\d|2[0-3]):([0-5]\d):([0-5]\d)$~', $replacement, $time_matches) === 1)
+				return sprintf('\'%02d:%02d:%02d\'', $time_matches[1], $time_matches[2], $time_matches[3]);
```

Comment:
That setting just controls how times are shown to the user. The code internally uses HH24:MM:SS for everything. 

Here is the flow:

1. The user enters a time on the form in any format that PHP understands. For example, `1:30 PM`.
2. The `setEventStartEnd()` function parses the input and converts it to HH24:MM:SS format. For example, `13:30:00`
3. The `insertEvent()` (or `modifyEvent()`) function sends `13:30:00` to the appropriate $smc function for insertion into the database.
4. The `smf_db_replacement__callback()` function checks the time string again to make sure it is in HH24:MM:SS format, and rewrites it in that format again, just to be extra sure.
5. The SQL call inserts the value into the database.

When the user wants to view an event, the HH24:MM:SS time string is retrieved from the database and reformatted for display using the setting you mentioned.